frontend/kesaseteli/employer/src/components/form/SubmitButton.tsx

```diff
@@ -0,0 +1,36 @@
+import { Button, IconArrowRight } from 'hds-react';
+import { getApplicationFormContext } from 'kesaseteli/employer/components/form/ApplicationForm';
+import Application from 'kesaseteli/employer/types/application';
+import React from 'react';
+
+type Props<T> = {
+  id: string;
+  onSubmit: (data: T) => void;
+  loadingText: string;
+  children: React.ReactNode;
+};
+
+const SubmitButton = ({
+  id,
+  loadingText,
+  onSubmit,
+  children,
+}: Props<Application>): ReturnType<typeof Button> => {
```

Comment:
You're right. Original idea was that the context would use generic type instead of Application but there were some problems and too limited time. I think it's still possible to do but perhaps in separate refactoring. With generic types the simple components and context could be moved to shared folder (if any other project would like to use react-hook-form components). For now I just removed the remainings of the generics :)